File: projects/9ee52fd7-13e2-4fa0-9830-79d5543e1e32/archive/20230815_140925/workspace/apirequest.py
import requests

class APIRequest:
    def __init__(self, url, headers):
        self.url = url
        self.headers = headers

    def get(self, path):
        response = requests.get(self.url + path, headers=self.headers)
        return response.json()

    def post(self, path, data):
        response = requests.post(self.url + path, headers=self.headers, json=data)
        return response.json()

File: projects/9ee52fd7-13e2-4fa0-9830-79d5543e1e32/archive/20230815_134247/workspace/ApiRequest.js
const axios = require('axios');

class ApiRequest {
  constructor(oauth2Client, url) {
    this.oauth2Client = oauth2Client;
    this.url = url;
  }

  async send(method, data) {
    const token = await this.oauth2Client.getAccessToken();

    const response = await axios({
      method,
      url: this.url,
      headers: {
        Authorization: `Bearer ${token}`
      },
      data
    });

    return response.data;
  }
}

module.exports = ApiRequest;
